### PR TITLE
Updating the cascade code, unit problems in SSC, p-synchrotron case-study

### PR DESCRIPTION
**Pull request by wrijupan**
This pull request have 3 major tasks -

1) First template for secondary cascade radiation : This avoids MC simulations following the approach of http://iopscience.iop.org/article/10.1088/0004-637X/768/1/54/pdf. The first generation of high energy gamma-rays and pairs from p-gamma interactions are calculated based on the analytical approximations by Kelner & Aharonian (2008). It is assumed that the injected secondary e+/e- pairs cool down predominantly via synchrotron.

2) Case study of proton synchrotron for PKS1510 2012 dataset

3) Fixing inconsistencies in units in the SSC code especially inside `numerics.py`

### CODE REVIEW

**Comment by cosimoNigro** [comment]
Thank you @wrijupan, this is a major improvement! 
I will review the changes tomorrow. From now on I suggest that you become responsible for the **hadronic** part, so any pull request related to that will be reviewed by you.

**Comment by wrijupan** [comment]
@cosimoNigro okay I will take responsibility for the hadronic part. Thanks!

File: SSC_test.py [hunk 1]
```diff
@@ -33,8 +33,8 @@
 IC = num.inverse_compton(N_e)
 
 stop = timeit.default_timer()
-print 'Computational time: '
-print stop - start, ' s'
+print ('Computational time: ')
+print (stop - start, ' s')
```
Comment (cosimoNigro) [inline SSC_test.py]: please include 
`from __future__ import division, print function` 
at the beginning of this kind of macro, in order to have compatibility with `python 2.7`

File: optical_depth/calc_opt_depth.py [hunk 2]
```diff
@@ -225,43 +218,47 @@ def integrand1(x): return self.phi_bar(x * w) / (x ** (4 - a))
         def integrand2(x): return self.phi_bar(x * w) / (x ** (4 - b))
         integral2 = quad(integrand2, max(1, 1 / w), x_b)[0]
 
-        return prefactor * (integral1 + integral2)
+        return float(prefactor * (integral1 + integral2))
 
 
 if __name__ == '__main__':
```
Comment (cosimoNigro) [inline optical_depth/calc_opt_depth.py]: What about having a test macro for the future, removing the `if __name__ == '__main__'` from the class and leaving it only as a collection of fucntions? We can also consider to use pytest. 
https://docs.pytest.org/en/latest/
For now it's ok. It is just a suggestion.

**Comment by cosimoNigro** [comment]
Once you correct my first comment I will merge it.

File: optical_depth/calc_opt_depth.py (hunk 2 above)
Comment (wrijupan) [reply optical_depth/calc_opt_depth.py]: I agree! I will do it in future.

**Pull request merged**